python-sdk/src/astro/files/locations/base.py

```diff
@@ -52,6 +52,24 @@ def size(self):
         """Return the size in bytes of the given file"""
         raise NotImplementedError
 
+    @property
+    @abstractmethod
+    def openlineage_dataset_namespace(self):
+        """
+        Returns the open lineage dataset namespace as per
+        https://github.com/OpenLineage/OpenLineage/blob/main/spec/Naming.md
+        """
+        raise NotImplementedError
```

Comment:
Why can we not implement this here instead of per location? It seems it looks the same for all locations, no?

Reply:
@feluelle the implementation would change based on each location as per https://github.com/OpenLineage/OpenLineage/blob/main/spec/Naming.md